aws-sigv4/src/lib.rs

```diff
@@ -103,17 +98,23 @@ impl Default for SigningSettings {
     fn default() -> Self {
         Self {
             uri_encoding: Double,
+            signed_body_header: SignedBodyHeaderType::NoHeader,
         }
     }
 }
 
+pub enum SignableBody<'a> {
+    Bytes(&'a [u8]),
+    UnsignedPayload,
+    Precomputed(String),
```

Comment:
Should this be a String or a `Vec<u8>`? Naively, most crypto code that I know shouldn't be validated for UTF-8.

Reply:
the intention is that this checksum had already been converted to hex but we could also have it be the raw bytes